code/go/0chain.net/smartcontract/storagesc/kill.go

```diff
@@ -0,0 +1,126 @@
+package storagesc
+
+import (
+	"strings"
+
+	"0chain.net/smartcontract/partitions"
+	"0chain.net/smartcontract/provider"
+	"0chain.net/smartcontract/stakepool"
+
+	cstate "0chain.net/chaincore/chain/state"
+	"0chain.net/chaincore/transaction"
+	"0chain.net/core/common"
+)
+
+func kill() {
+
+}
+
+// killBlobber
+// punitively disables a blobber. it will no longer be used for new allocations
+// or receive further rewards. Stakeholders will have their stakes slashed.
+func (_ *StorageSmartContract) killBlobber(
+	tx *transaction.Transaction,
+	input []byte,
+	balances cstate.StateContextI,
+) (string, error) {
+	conf, err := getConfig(balances)
+	if err != nil {
+		return "", common.NewError("can't get config", err.Error())
+	}
+
+	var blobber = newBlobber("")
```

Comment:
why not `&Blobber{}`

Reply:
Why not not? Initially, it was done this way because I embedded `*Provider` into StorageNode, this was changed to embed `Provider` so either way now works. I left the code as it was so that if later StorageNode needs default initialisation it can all be done in one place.

Reply:
it's a natural way for golang to use default initialization `&Blobber{}` when possible, it's more obvious what happens when you read code, imagine you are reading code and see `newBlobber("")` do you know what will happen inside this func?